### ISSUE TITLE
How to add a door to a wall?

### ISSUE THREAD

**Issue by dbojadievski** [xBimTeam/XbimEssentials#49]
I am experimenting with your toolkit, and it so far seems very useful. However, due to a lack of clear documentation, I'm struggling to understand how to compose hierarchies of objects. 

For an example, let's assume I would like to create a wall, that has a door in it. Creating both the wall and the door themselves is fairly straightforward, after creating a model itself. But, how would one go about attaching the door to the wall?

**Comment by martin1cerny** [comment]
You need to create opening object and build the hierarchy. See [this](http://www.buildingsmart-tech.org/ifc/IFC4/Add1/html/link/ifcopeningelement.htm).

**Comment by dbojadievski** [comment]
I understood the relationship mechanism. Thank you very much.

**Comment by FrieBIM91** [comment]
I am also trying to add a window to a wall. According to xBIM Xplorer the semantic hierarchy is correct already.
I created an IfcOpeningStandardCase, IfcRelVoidsElement, IfcRelFillsElement and the relations.

I am still struggling with the geometry for a few days now...
Could you post a code snippet how you put a window/door in the middle of a wall?

Thanks a lot.
@dbojadievski @martin1cerny

**Comment by martin1cerny** [comment]
I'm sorry but this is beyond the level of support I can provide here. I can only recommend you to create very basic example (simple wall with an empty opening, wall with very simple window) in some authoring tool (Revit, Archicad, Tekla etc.) and study the structure of the file. Make sure your placements and their relations are correct.

**Comment by FrieBIM91** [comment]
Thanks for your fast response. I managed to add an IfcWall + IfcOpening + IfcWindow to an existing IFC-file (IFC4).
Unfortunately this didn't work for an IFC2x3, although I implemented the "Fast upgrade track to IFC version agnostic solution". I had to repeat the same process (with IFC2x3 libraries) in order to get it work.

Is there a better approach for an IFC-Schema independent solution? 

Thanks a lot.

**Comment by martin1cerny** [comment]
Please, take the time to read the [more detailed description](http://docs.xbim.net/xbim-4/xbim-4-release-notes.html) which includes use of `Xbim.Ifc4.Interfaces.Create` helper object.

**Comment by FrieBIM91** [comment]
Sorry its me again. Of course I did. I created a Create Instance in CreateWall() in the "Create Proper Wall in 3D" Example and created all non-abstract entities via create.Wall(), create.CartesianPoint(), and so on. Thus of course some functions are not working anymore. For instance IfcCartesianPoint.SetXYZ()
 
 Furthermore according to the changes from IfcWall to IIFcWall I can not add easily the wall to a BuildingStorey anymore.
